server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -79,11 +79,18 @@ public boolean isContentConsumed() {
         return contentConsumed;
     }
 
+    // for testing
```

Comment:
also if we make mediatypeparser even more strict, we could remove the RestRequest.parseContentType (line 504) and line 60
WDYT?